File: generate-city-pages.js
const fs = require('fs');
const path = require('path');

const cities = [
  { name: 'Leiden', slug: 'leiden', gpsLat: '52.1601', gpsLon: '4.4970' },
  { name: 'Zoetermeer', slug: 'zoetermeer', gpsLat: '52.0575', gpsLon: '4.4937' },
  { name: 'Dordrecht', slug: 'dordrecht', gpsLat: '51.8133', gpsLon: '4.6901' },
  { name: 'Delft', slug: 'delft', gpsLat: '52.0116', gpsLon: '4.3571' },
  { name: 'Gouda', slug: 'gouda', gpsLat: '52.0115', gpsLon: '4.7108' },
  { name: 'Schiedam', slug: 'schiedam', gpsLat: '51.9186', gpsLon: '4.3997' },
  { name: 'Alphen aan den Rijn', slug: 'alphen-aan-den-rijn', gpsLat: '52.1285', gpsLon: '4.6573' },
  { name: 'Vlaardingen', slug: 'vlaardingen', gpsLat: '51.9122', gpsLon: '4.3419' },
  { name: 'Rijswijk', slug: 'rijswijk', gpsLat: '52.0367', gpsLon: '4.3208' },
  { name: 'Capelle aan den IJssel', slug: 'capelle-aan-den-ijssel', gpsLat: '51.9289', gpsLon: '4.5778' },
  { name: 'Rotterdam', slug: 'rotterdam', gpsLat: '51.9225', gpsLon: '4.47917' }
];

// Read templates
const computerhulpTemplate = fs.readFileSync('./app/computerhulp-aan-huis/page.tsx', 'utf8');
const studentTemplate = fs.readFileSync('./app/student-aan-huis/page.tsx', 'utf8');

cities.forEach(city => {
  // Generate computerhulp page
  let computerhulpContent = computerhulpTemplate
    .replace(/Zuid-Holland/g, city.name)
    .replace(/zuid-holland/g, city.slug)
    .replace(/computerhulp-aan-huis'/g, `computerhulp-aan-huis-${city.slug}'`)
    .replace(/computerhulp-aan-huis"/g, `computerhulp-aan-huis-${city.slug}"`)
    .replace(/computerhulp-aan-huis`/g, `computerhulp-aan-huis-${city.slug}\``)
    .replace(/Computerhulp aan Huis'/g, `Computerhulp aan Huis ${city.name}'`)
    .replace(/latitude: '52.0705'/g, `latitude: '${city.gpsLat}'`)
    .replace(/longitude: '4.3007'/g, `longitude: '${city.gpsLon}'`)
    .replace(/addressRegion: 'Zuid-Holland'/g, `addressLocality: '${city.name}', addressRegion: 'Zuid-Holland'`);

  const computerhulpDir = `./app/computerhulp-aan-huis-${city.slug}`;
  if (!fs.existsSync(computerhulpDir)) {
    fs.mkdirSync(computerhulpDir, { recursive: true });
  }
  fs.writeFileSync(path.join(computerhulpDir, 'page.tsx'), computerhulpContent);

  // Generate student page
  let studentContent = studentTemplate
    .replace(/Zuid-Holland/g, city.name)
    .replace(/zuid-holland/g, city.slug)
    .replace(/student-aan-huis'/g, `student-aan-huis-${city.slug}'`)
    .replace(/student-aan-huis"/g, `student-aan-huis-${city.slug}"`)
    .replace(/student-aan-huis`/g, `student-aan-huis-${city.slug}\``)
    .replace(/Student aan Huis'/g, `Student aan Huis ${city.name}'`)
    .replace(/latitude: '52.0705'/g, `latitude: '${city.gpsLat}'`)
    .replace(/longitude: '4.3007'/g, `longitude: '${city.gpsLon}'`)
    .replace(/addressRegion: 'Zuid-Holland'/g, `addressLocality: '${city.name}', addressRegion: 'Zuid-Holland'`);

  const studentDir = `./app/student-aan-huis-${city.slug}`;
  if (!fs.existsSync(studentDir)) {
    fs.mkdirSync(studentDir, { recursive: true });
  }
  fs.writeFileSync(path.join(studentDir, 'page.tsx'), studentContent);

  console.log(`✓ Generated pages for ${city.name}`);
});

console.log('\n✓ All 22 city pages generated successfully!');

File: generate_city_pages.py
#!/usr/bin/env python3
# Script to generate all city pages with exact template structure

import os
import re

# City data for computerhulp pages
computerhulp_cities = [
    {"name": "Zoetermeer", "gps": "52.0575, 4.4937", "wikidata": "Q9927", "url_name": "zoetermeer"},
    {"name": "Dordrecht", "gps": "51.8133, 4.6901", "wikidata": "Q26421", "url_name": "dordrecht"},
    {"name": "Delft", "gps": "52.0116, 4.3571", "wikidata": "Q33433", "url_name": "delft"},
    {"name": "Gouda", "gps": "52.0115, 4.7108", "wikidata": "Q33456", "url_name": "gouda"},
    {"name": "Schiedam", "gps": "51.9186, 4.3997", "wikidata": "Q34047", "url_name": "schiedam"},
    {"name": "Alphen aan den Rijn", "gps": "52.1285, 4.6573", "wikidata": "Q213061", "url_name": "alphen-aan-den-rijn"},
    {"name": "Vlaardingen", "gps": "51.9122, 4.3419", "wikidata": "Q9920", "url_name": "vlaardingen"},
    {"name": "Rijswijk", "gps": "52.0367, 4.3208", "wikidata": "Q9920", "url_name": "rijswijk"},
    {"name": "Capelle aan den IJssel", "gps": "51.9289, 4.5778", "wikidata": "Q33422", "url_name": "capelle-aan-den-ijssel"},
]

# Student cities (all 11 including Rotterdam)
student_cities = [
    {"name": "Rotterdam", "gps": "51.9225, 4.47917", "wikidata": "Q34370", "url_name": "rotterdam"},
    {"name": "Leiden", "gps": "52.1601, 4.4970", "wikidata": "Q43631", "url_name": "leiden"},
    {"name": "Zoetermeer", "gps": "52.0575, 4.4937", "wikidata": "Q9927", "url_name": "zoetermeer"},
    {"name": "Dordrecht", "gps": "51.8133, 4.6901", "wikidata": "Q26421", "url_name": "dordrecht"},
    {"name": "Delft", "gps": "52.0116, 4.3571", "wikidata": "Q33433", "url_name": "delft"},
    {"name": "Gouda", "gps": "52.0115, 4.7108", "wikidata": "Q33456", "url_name": "gouda"},
    {"name": "Schiedam", "gps": "51.9186, 4.3997", "wikidata": "Q34047", "url_name": "schiedam"},
    {"name": "Alphen aan den Rijn", "gps": "52.1285, 4.6573", "wikidata": "Q213061", "url_name": "alphen-aan-den-rijn"},
    {"name": "Vlaardingen", "gps": "51.9122, 4.3419", "wikidata": "Q9920", "url_name": "vlaardingen"},
    {"name": "Rijswijk", "gps": "52.0367, 4.3208", "wikidata": "Q9920", "url_name": "rijswijk"},
    {"name": "Capelle aan den IJssel", "gps": "51.9289, 4.5778", "wikidata": "Q33422", "url_name": "capelle-aan-den-ijssel"},
]

print(f"Computerhulp cities: {len(computerhulp_cities)}")
print(f"Student cities: {len(student_cities)}")
print(f"Total to generate: {len(computerhulp_cities) + len(student_cities)}")
